### ISSUE TITLE
[Feature Request] Compare hash after download

### ISSUE THREAD

**Issue by hazre** [hazre/neosvr-mod-manager#3]
requested by ljoonal#2847 in NML [discord](https://canary.discord.com/channels/901126079857692714/1041151587508289607/1041454360955195392).

**Comment by ljoonal** [comment]
Yup, the [manifest design docs](https://www.neosmodloader.com/manifest-design) outlines the basic goals & security concerns with the manifest, and the most relevant to installer/updater tools is:

> Github artifacts are NOT immutable, meaning a fixed URL does not guarantee a fixed file. Therefore we MUST verify artifacts clientside.

Whilst it's not really a concern with just listing the mods, providing an easy to use way to install them whilst not verifying that the hashes match fully voids the [modding community auditing process](https://www.neosmodloader.com/auditing-process). I feel like it should be stated or fixed, since this might give users a false sense of security with the way that [neosmodloader.com](https://www.neosmodloader.com/) is linked to, since the landing page states that:

> The mod list is a community-managed listing of mods that are vetted to not be malicious.
> The purpose of the list is to make it easier & safer for people to run mods. Since mods are another form of executables, and it’s generally not recommended to not run random executables from the internet.

**Comment by hazre** [comment]
Notes to myself:

Possible ways of displaying a user prompt when hashes don't match:
- [Window.confirm()](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm) (Vanilla API but ugly)
- [Svelte Component API](https://svelte.dev/repl/15d7c21aa036464cb0220af60dde3843?version=3.48.0) (Customizable but requires a refactor of download function)
